File: web/src/main/kotlin/no/ssb/kostra/web/viewmodel/ErrorReportVm.kt
package no.ssb.kostra.web.viewmodel

import io.micronaut.core.annotation.Introspected

@Introspected
data class ErrorReportVm(
    val innparametere: ReportRequestVm,
    val antallKontroller: Int,
    val feilkode: KostraErrorCode,
    val feil: Collection<ErrorDetailsVm>
)

File: web/src/main/vite-project/src/kostratypes/errorReportVm.ts
import {KostraFormVm} from "./kostraFormVm";
import {ErrorDetailsVm} from "./errorDetailsVm";
import {KostraErrorCode} from "./kostraErrorCode";

export interface ErrorReportVm {
    readonly innparametere: NonNullable<KostraFormVm>
    readonly antallKontroller: NonNullable<number>
    readonly feilkode: NonNullable<KostraErrorCode>
    readonly feil: ErrorDetailsVm[]
}